src/extension_utils/utils.ts

```diff
@@ -372,9 +418,194 @@ export const installPythonDependencies = async (
+                        open(CONSTANTS.LINKS.INSTALL);
+                    }
+                });
+        }
+
         console.error(err);
         this.logToOutputChannel(errorChannel, err.toString(), true /* show */);
-        installed = false;
+        return false;
     }
-    return installed;
+};
+
+export const setupEnv = async (
+    context: vscode.ExtensionContext,
+    needsResponse: boolean = false
+) => {
+    let pythonExecutableName = "";
+    let originalPythonExecutableName = "";
+
+    // try to get name from interpreter
+    try {
+        originalPythonExecutableName = getConfig(CONFIG.PYTHON_PATH);
+    } catch (err) {
+        originalPythonExecutableName = "python";
+    }
+
+    if (
+        originalPythonExecutableName === "python" ||
+        originalPythonExecutableName === ""
+    ) {
+        try {
+            const { stdout } = await exec(
+                'python -c "import sys; print(sys.executable)"'
+            );
+            originalPythonExecutableName = stdout.trim();
+        } catch (err) {
+            vscode.window
+                .showErrorMessage(
+                    CONSTANTS.ERROR.NO_PYTHON_PATH,
+                    DialogResponses.INSTALL_PYTHON
+                )
+                .then((selection: vscode.MessageItem | undefined) => {
+                    if (selection === DialogResponses.INSTALL_PYTHON) {
+                        const okAction = () => {
+                            open(CONSTANTS.LINKS.DOWNLOAD_PYTHON);
+                        };
+                        showPrivacyModal(
+                            okAction,
+                            CONSTANTS.INFO.THIRD_PARTY_WEBSITE_PYTHON
+                        );
+                    }
+                });
+            return "";
+        }
+    }
+    // fix path to be absolute
+    if (!path.isAbsolute(originalPythonExecutableName)) {
+        originalPythonExecutableName = path.join(
+            vscode.workspace.rootPath,
+            originalPythonExecutableName
+        );
+    }
+
+    if (!fs.existsSync(originalPythonExecutableName)) {
+        await vscode.window.showErrorMessage(CONSTANTS.ERROR.BAD_PYTHON_PATH);
+        return "";
+    }
+
+    if (!(await validPythonVersion(originalPythonExecutableName))) {
+        return "";
+    }
+
+    pythonExecutableName = originalPythonExecutableName;
+
+    if (!(await checkForDependencies(context, pythonExecutableName))) {
+        // environment needs to install dependencies
+        if (!(await checkIfVenv(context, pythonExecutableName))) {
+            pythonExecutableName = await getPythonVenv(context);
+            if (await hasVenv(context)) {
+                // venv in extention exists with wrong dependencies
+                if (
+                    !(await checkForDependencies(context, pythonExecutableName))
+                ) {
+                    await installDependencies(context, pythonExecutableName);
```

Comment:
Do we want to do anything if this fails?

Reply:
installDependencies has a try-catch block inside, so it should never throw an error. The only things outside of the try-catch block in installDependencies is getPathToScript (which should never fail) and checkForPip (which also has its own try-catch)

Reply:
Doesn't it return false if pip is not installed? I was more thinking on these lines

Reply:
fixed in the commit! Will throw an error if dependency download fails :)